File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        M = max(candies)
        lst = [x-M+extraCandies for x in candies]
        result=[0]*len(candies)
        for i in range(len(lst)):
            if lst[i]>=0:
                result[i]=True
            else:
                result[i]=False
        return result
            
              

File: 1528-kids-with-the-greatest-number-of-candies/kids-with-the-greatest-number-of-candies.c

bool* kidsWithCandies(int* candies, int candiesSize, int extraCandies, int* returnSize) {
    int M = 0;
    for (int i=0; i<candiesSize;i++){
        if(candies[i]>M) M=candies[i];
    }
    for (int i=0; i<candiesSize;i++){
        candies[i]=candies[i]-M+extraCandies;
    }
    bool* result = malloc((candiesSize)*sizeof(bool));
    for (int i=0; i<candiesSize;i++){
        if(candies[i]>=0) result[i]=true;
        else result[i]=false;
    }
    *returnSize=candiesSize;
    return result;

}